stage1/diagexec/diagexec.c

```diff
@@ -221,12 +265,14 @@ int main(int argc, char *argv[])
 		NULL
 	};
 
-	const char *root, *cwd, *env, *uid_str, *gid_str, *exe;
-	char **args;
-	uid_t uid;
-	gid_t gid;
+	const char	*root, *cwd, *env, *uid_str, *gid_str, *exe;
+	char		**args;
+	uid_t		uid;
+	gid_t		*gids;
+	size_t		n_gids;
```

Comment:
Shouldn't `n_gids` be an `int` instead of a `size_t`? It's the number of gids, not `number * sizeof(gid_t)`.

Ditto in the prototype of the function `parse_gids`.